examples/pipe-open/client.rs

```diff
@@ -34,27 +36,22 @@ pub struct PipeClient {
 impl PipeClient {
     /// Creates a new pipe client.
     pub fn new(libos: LibOS, pipe_name: String) -> Result<Self> {
-        Ok(Self { libos, pipe_name })
+        Ok(Self {
+            libos,
+            pipe_name,
+            qds: Vec::default(),
+        })
     }
 
     // Runs the target pipe client.
     pub fn run(&mut self, niterations: usize) -> Result<()> {
-        let mut qds: Vec<QDesc> = Vec::default();
-
         // Open several pipes.
         for i in 0..niterations {
             let pipeqd: QDesc = self.libos.open_pipe(&format!("{}:rx", self.pipe_name))?;
-            qds.push(pipeqd);
-            // Clean up open pipes on error
-            // FIXME: https://github.com/demikernel/demikernel/issues/638
+            self.qds.push(pipeqd);
             self.push_and_wait(pipeqd, 1, i as u8)?;
         }
 
-        // Close all TCP pipes.
-        for qd in qds {
-            self.libos.close(qd)?;
-        }
```

Comment:
Should the qds still be closed in run()? This is the earliest opportunity to free up resources. Also, then we can treat drop() as a safety mechanism instead of relying on it for cleanup.